vendor/github.com/google/cadvisor/fs/fs.go

```diff
@@ -533,6 +533,21 @@ func (self *RealFsInfo) GetDirFsDevice(dir string) (*DeviceInfo, error) {
 	}
 
 	mount, found := self.mounts[dir]
+	// try the parent dir if not found until we reach the root dir
+	// this is an issue on btrfs systems where the directory is not
+	// the subvolume
+	for !found {
```

Comment:
The same comment I raised in another cherrypick pr, but thought it is worthy of mentioning it again here: The comment says that this is to fix btrfs issue, but the code logic added here isn't capped for btrfs only. Is there any potential issue on other systems?

Reply:
other systems should have already returned at this point.  The `if` statement after this is only for brtfs, so any other system will return the error at the end of the function.